app/models/grant_submission/response.rb

```diff
@@ -119,6 +120,10 @@ def submitted?
       self.submission.submitted? && (!self.submission.changed? || self.submission.changes.keys.include?('state'))
     end
 
+    def changed_for_autosave?
+      super || document.changed_for_autosave? if document.attached?
+    end
+
```

Comment:
with the update in `file_upload_field`, I'd be curious whether this is still necessary.